src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -510,4 +511,22 @@ close(int dev_handle) const
   pcidev::get_dev(get_device_id(), false)->close(dev_handle);
 }
 
+void
+device_linux::
+load_xclbin(std::vector<char> &buffer) const {
```

Comment:
Being that std::vector<char> is being used in a lot of places and, have you thought about defining it in a "using" statement.  Something like:
`using MemoryBuffer=std::vector<char>;`
and then changing this method signature to:
`load_xclbin(const MemoryBuffer &buffer) const {`

And yes, the buffer should be declared `const `for it isn't being changed.

Reply:
Hmm. I think leaving it as a vector should be fine. `std::vector<char>` isn't verbose and is being used 1 or 2 times per file. 

I can't define the buffer as const since we are casting it to `xclmgmt_ioc_bitstream_axlf` in L 521

Reply:
Hmm, that last time I checked, this buffer is not being modified, am I mistakened?

If not, then this buffer should be a const and xclbmgmt_ioc_bistream_axlf should be updated to take a const value.   

I understand that that might be be possible to update ALL of the locations in a timely manner, as such, it might be best to use a const_cast() of the buffer when calling this method.

As for the std::vector<char> it would be better it is was aliased.   You can consider this as part of the C++ coding standard that we have started to adopt.  In this case, the alias is used to simplify notation and hide implementation details.